presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java

```diff
@@ -43,7 +43,7 @@
     private final Optional<List<Type>> types;
     private final Optional<List<String>> clusteredBy;
     private final Optional<List<Integer>> clusterCount;
-    private final Optional<List<String>> distribution;
+    private final Optional<List<Object>> distribution;
```

Comment:
Well, as discussed offline, we will need a good data structure to represent distribution. `ColumnStatistics` is a good example. We likely could make it a thrift serializable class. This part needs some brainstorming. Let's chat offline.